pkgs/development/interpreters/python/hooks/setuptools-legacy-namespace-hook.sh

```diff
@@ -0,0 +1,28 @@
+# Setup hook for setuptools.
+echo "Sourcing setuptools-legacy-namespace-hook"
+
+setuptoolsLegacyNamespaceFixupPhase() {
+    echo "Executing setuptoolsLegacyNamespaceFixupPhase"
+
+    # We are going to delete every *-nspkg.pth file, which is generated by
+    # setuptools when a package uses the legacy namespace package code path
+    # (when using the namespace_packages keyword in setup.{cfg,py}).
+    #
+    # These files can cause subtle problems when used alongside packages that
+    # use a more modern code path that no longer generates those files. See:
+    #
+    #   https://github.com/pypa/setuptools/discussions/3991
+    #
+    SETUPTOOLS_NSPKG_PTH_FILES="$(find "$out/lib" -type f -name '*-nspkg.pth')"
+    if [[ -n "$SETUPTOOLS_NSPKG_PTH_FILES" ]]; then
+      find "$out/lib" -type f -name '*-nspkg.pth' -delete
```

Comment:
```suggestion
      rm $SETUPTOOLS_NSPKG_PTH_FILES
```

Reply:
Calling `find` twice is less efficient, but I did it on purpose just in case the hook finds more than one `-nspkg.pth` file. The second `find` call handles deletion properly even if there are spaces in the file name. I can also limit the code to only deleting the first file found, or leave a better comment. What do you think?

Reply:
Can't we tell find to exit with an error if it can't find files and then delete them right away?